### PR TITLE
Implement events

### PR DESCRIPTION
**Pull request by saintedlama**
Events implementation adapted to standard code style. The split of getting a server topology and connecting to a server also enables us to implement ssl authentication in the right place - the `connect` module.

### CODE REVIEW

**Comment by saintedlama** [comment]
@watson @sorribas @mafintosh No more WIP but ready to merge

File: lib/database.js [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+var getTopology = require('./get-topology')
+var connect = require('./connect')
 
 var noop = function () {}
 
-var Database = function (opts, onserver) {
-  this._getServer = onserver
-  this._dbname = opts.name
-
+var Database = function (db, options) {
```
Comment (watson) [inline lib/database.js]: I think it's confusing that the 1st argument is called `db` as one would think it's an instance of some higher order database-thingy, where in fact it's just an options hash (as far as I read the code at least).

But if that's the case then we suddenly have two options arguments, which is a bit confusing for the reader.

So if I understand this correctly, I think the best solution would be to merge the two and just have a single options argument. In the index.js file where you call the `Database` constructor you could then just add the `connString` and the `cols` to the `options` hash parsed in by the user, e.g:

```js
var db = new Database(extend({ connString: connString, cols: cols }, options))
```

Comment (saintedlama) [inline lib/database.js]: Good point! fixed.

**Comment by watson** [comment]
:+1:

**Comment by saintedlama** [comment]
@mafintosh had to merge the pull request because with one additional commit this let's us anwers #221 (Does mongojs support SSL/TLS connections?) with YES :dancers:

Implementation was dead simple :beers:

**Pull request merged**